documentation/src/docs/metrics.md

```diff
@@ -58,15 +97,17 @@ public interface OpStatsLogger {
 }
 ```
 
-- reportSuccessEvent() : Used to track Timer of a successful operation and will record the latency in Nanoseconds in required metric. 
-- reportFailEvent() : Used to track Timer of a failed operation and will record the latency in Nanoseconds in required metric.  
-- reportSuccessValue() : Used to track Histogram of a success value.
-- reportFailValue() : Used to track Histogram of a failed value. 
-- toOpStatsData() :  Used to support JMX exports and inner test.
-- clear : Used to clear stats for this operation.
+- `reportSuccessEvent()`: It is used to track the [Timer](http://metrics.dropwizard.io/3.1.0/manual/core/#timers) of a successful operation and will record the latency in nanoseconds in the required metric.
+- `reportFailEvent()`: It is used to track the [Timer](http://metrics.dropwizard.io/3.1.0/manual/core/#timers) of a failed operation and will record the latency in nanoseconds in required metric.  
+- `reportSuccessValue()`: It is used to track the [Histogram](http://metrics.dropwizard.io/3.1.0/manual/core/#histograms) of a success value.
+- `reportFailValue()`: It is used to track the [Histogram](http://metrics.dropwizard.io/3.1.0/manual/core/#histograms) of a failed value.
+- `toOpStatsData()`:  It is used to support the [JMX](https://metrics.dropwizard.io/3.1.0/manual/core/#jmx) exports and inner tests.
+- `clear`: It is used to clear the stats for this operation.
+
+## Metric Logger — Interface DynamicLogger
+
+The following is an example of a  simple interface that exposes only simple type metrics: ([Counter](https://metrics.dropwizard.io/3.1.0/manual/core/#counters)/[Gauge](https://metrics.dropwizard.io/3.1.0/manual/core/#gauges)/[Meter](https://metrics.dropwizard.io/3.1.0/manual/core/#meter)).
 
-### 1.4 Metric Logger — interface DynamicLogger
-A simple interface that only exposes simple type metrics: Counter/Gauge/Meter.
 ```java
 public interface DynamicLogger {
```

Comment:
Can you please remove all these examples? Please include references to actual code files. We should not be having copy-pasted code in this document.

Reply:
This comment applies to all these code blocks, not just this one.

Reply:
@andreipaduroiu `public interface Dynamic Logger` sounds similar to `https://github.com/pravega/pravega/blob/43113a257681c92581d57c6ed34f7b9e36e0bfe3/shared/metrics/src/main/java/io/pravega/shared/metrics/DynamicLogger.java` , i hope i can retain this one.

Reply:
@Aparnarr  Yes, you're on the right path. However your link points to a Pull Request, which is not desirable. When you go to the Pravega GitHub main page, click `Code` on the top left and access the source code tree path from there.

Alternatively (I guess you found this via a search box), click the `Copy Path` button on the top-right of that page. It should copy the URL to the file in the source code. 

Doing it for this file gave me: `pravega/shared/metrics/src/main/java/io/pravega/shared/metrics/DynamicLogger.java`. You can just prefix this with `https://github.com/` and finish it.     

PS: the way you can tell if the URL is pointing to a commit or PR is if you see a UUID in it (`/blob/` followed by a sequence of letters and numbers).

Reply:
Let's fix this for the rest of the links too. Otherwise it looks much better.

Reply:
Done. All the hyperlinks modified